common/src/main/java/software/tnb/common/utils/HTTPUtils.java

```diff
@@ -74,14 +79,19 @@ public void delete(String url, Map<String, String> headers) {
         execute(new Request.Builder().url(url).delete().headers(Headers.of(headers)).build(), true);
     }
 
-    private Response execute(Request request, boolean throwError) {
+    private Response execute(Request request, boolean throwError, int attempts) {
         try {
             okhttp3.Response response = client.newCall(request).execute();
             String responseBody = null;
             if (response.body() != null) {
                 responseBody = response.body().string();
                 response.body().close();
             }
+            if (withRetry && attempts > 0 && retryAllowedMethods.contains(request.method()) && this.retryCodes.contains(response.code())) {
+                LOG.warn("retrying the http call in 1 second");
+                WaitUtils.sleep(1000);
+                execute(request, throwError, attempts - 1);
```

Comment:
shouldn't there be a `return execute..`?

Reply:
sure, thanks!